docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,259 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream methods for downloading files, it is time to introduce you to additional *upstream* methods. Upstream methods refers to functions that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to manage a repository with Git-like commands.
+
+## `HfApi`
+
+The `HfApi` class is a low level class that wraps around HTTP requests. There are many useful tasks you can accomplish with the `HfApi` class including: 
+
+- List and filter models.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models, and filter those models according to a specific language or framework. Use the `list_models` function with the `filter` parameter to search for a model of interest:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List all models.
+>>> api.list_models()
+
+# List only text classification models.
+>>> api.list_models(filter="text-classification")
+
+# List only Russian models compatible with PyTorch.
+>>> api.list_models(filter=("ru", "pytorch"))
+
+# List only the models trained on the "common_voice" dataset.
+>>> api.list_models(filter="dataset:common_voice")
+
+# List only the models from the spaCy library.
+>>> api.list_models(filter="spacy")
+```
+
+Explore available public datasets with `list_datasets`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List only text classification datasets.
+>>> api.list_datasets(filter="task_categories:text-classification")
+
+# List only datasets in Russian for language modeling.
+>>> api.list_datasets(filter=("languages:ru", "task_ids:language-modeling"))
+```
+
+### Inspect model or dataset metadata
+
+Get important information about a model or dataset as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# Get metadata of a single model.
+>>> api.model_info("distilbert-base-uncased")
+
+# Get metadata of a single dataset.
+>>> api.dataset_info("glue")
+```
+
+### Delete a repository
```

Comment:
This section is great. I think we should have a section for `create_repo` as well

Reply:
In the tutorial (`tutorial-add-library`), I have a section for `create_repo`. I didn't introduce `delete_repo` in the tutorial because it seems like more like a repo management tool. But let me know if you think it'd make more sense to include `delete_repo` in the tutorial!

Reply:
I think it is quite intuitive to have `create_repo` and `delete_repo` side-by-side, so would also favour linking them